lib/modem_jwt/modem_jwt.c

```diff
@@ -148,6 +158,43 @@ int modem_jwt_generate(struct jwt_data *const jwt)
 	return ret;
 }
 
+const char *modem_jwt_get_uuid()
+{
+	struct jwt_data jwt = {0};
+
+	int rc = modem_jwt_generate(&jwt);
+	if (rc) {
+		modem_jwt_free(jwt.jwt_buf);
+		return NULL;
+	}
+
+	char *p = strchr(jwt.jwt_buf, '.') + 1;
+	char *end = strrchr(jwt.jwt_buf, '.');
+	char *payload =  k_calloc(1, 256);
```

Comment:
you could use a macro such as:
https://github.com/nrfconnect/sdk-nrf/blob/8f5922cb5cd660612975d8226e5b52d3d318cce7/lib/modem_attest_token/modem_attest_token.c#L18

to determine the proper  base64 payload size